server/service/testing_utils.go

```diff
@@ -202,14 +204,16 @@ func newTestServiceWithClock(t *testing.T, ds fleet.Datastore, rs fleet.QueryRes
 
 func createTestUsers(t *testing.T, ds fleet.Datastore) map[string]fleet.User {
 	users := make(map[string]fleet.User)
-	userID := uint(1)
-	for _, u := range testUsers {
+	// Map iteration is random so we sort and iterate using the testUsers keys.
+	keys := maps.Keys(testUsers)
+	sort.StringSlice(keys).Sort()
```

Comment:
instead of importing a new dependency, can we do this?
```
var keys []string
	for key := range testUsers {
		keys = append(keys, key)
	}
	sort.Strings(keys)
```